src/posix/platform/netif.cpp

```diff
@@ -450,10 +452,16 @@ static void UpdateUnicastLinux(otInstance *aInstance, const otIp6AddressInfo &aA
 
     req.ifa.ifa_family    = AF_INET6;
     req.ifa.ifa_prefixlen = aAddressInfo.mPrefixLength;
-    req.ifa.ifa_flags     = IFA_F_NODAD;
+    req.ifa.ifa_flags     = 0; // This value is overwritten by the IFA_FLAGS attribute in the Netlink message.
     req.ifa.ifa_scope     = aAddressInfo.mScope;
     req.ifa.ifa_index     = gNetifIndex;
 
+    // Don't add fe80::/64 route to prevent host processes sending traffic to Thread network interface by accident.
+    if (aIsAdded && aAddressInfo.mScope == kLinkLocalScope)
```

Comment:
With this change, can host still have link-local communication over Thread?

Reply:
No. Is there a use case?

Reply:
Per offline discussion, changed the PR to allow configuring the metric of the route.